buildSrc/src/main/kotlin/JvmToolchain.kt

```diff
@@ -0,0 +1,14 @@
+import org.gradle.jvm.toolchain.JavaLanguageVersion
+import org.gradle.jvm.toolchain.JvmVendorSpec
+import org.gradle.api.Project
+
+private const val JDK_VERSION = 11
+
+val jvmToolchainVersion: JavaLanguageVersion
+    get() {
+        val jvmVersion = System.getProperty("PKL_JVM_VERSION", JDK_VERSION.toString()).toInt()
```

Comment:
I believe that in Gradle, it is more conventional to use project properties rather than system properties to specify this kind of build parameters. I'd write this as
```kotlin
private const val DEFAULT_JDK_VERSION = 11

val Project.jvmToolchainVersion: JavaLanguageVersion
    get() {
        val jvmVersion = if (extra.has("pklJvmVersion")) extra.get("pklJvmVersion")?.toInt() else null
        return JavaLanguageVersion.of(jvmVersion ?: DEFAULT_JDK_VERSION)
    }
```

Then you'd run the build as `./gradlew check -PpklJvmVersion=17` (`-P` vs `-D`).

Reply:
Sorry, which discussion it is? I'm not quite sure I understand the reasoning.

Unfortunately, Gradle docs on various kinds of properties do not provide an explicit prescription on where to use which kind of thing, but they are interchangeable in a way - you can define system properties via project properties (`-PsystemProp.a.b.c=value`) and vice versa (`-Dorg.gradle.project.a.b.c=value`).

So, from my experience, I suggest using project properties instead of system properties, because:
1. This mechanism is a part of the Gradle API, and thus project properties can be retrieved in a variety of useful ways (like `val someProp by project.extra` delegates)
2. As a consequence of 1, project properties are used more in idiomatic Gradle code than system properties, in my experience.

Reply:
But overall I don't think I have a very strong stance on it, that was just a suggestion about what I think is a best practice. As I said, system and project properties are mostly interchangeable in Gradle, so the primary difference is the API which is used to access them, and arguably without delegates, system properties API is marginally nicer since it does not throw exceptions)

Reply:
The PR started with a Gradle property.
@translatenix suggested [here](https://github.com/apple/pkl/pull/262#issuecomment-1969963577) to use an System property:

> I think the project property is only useful for command-line invocations, so it could be a system property.

And followed up [here](https://github.com/apple/pkl/pull/262#issuecomment-1971318196) with:

> System property would be a small improvement because setting the project property in a build script will likely not work correctly.

Since I don't have a strong stance about it either, I changed it 🤷 

because you (@netvl) say the following, I'm going to resolve this conversation 🙃 
> But overall I don't think I have a very strong stance on it, that was just a suggestion about what I think is a best practice.